File: Week 2- Arrays/readbility/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Please Enter text: ");


    double letters = count_letters(text);
    double words = count_words(text);
    double sentences = count_sentences(text);

    double L = letters / words * 100;
    double S = sentences / words * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    // Compute number of letters and return as int
    int length = strlen(text);
    int letter_count = 0;
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letter_count++;
        }
    }
    return letter_count;
}

int count_words(string text)
{
    //compute number of word and return as int
    int length = strlen(text);
    int word_count = 1;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == 32)
        {
            word_count++;
        }
    }
    return word_count;
}

int count_sentences(string text)
{
    //Compute number of sentences and return as int
    int length = strlen(text);
    int sentence_count = 0;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence_count++;
        }
    }
    return sentence_count;
}

File: Week 6 - Python/sentimental-readability/readability.py
from cs50 import get_string


def main():
    '''uses Coleman-Liau index to grade difficulty of a text'''

    # Ask user for text
    text = get_string("Please enter text:\n")

    # Calculate L and S
    words = count_words(text)
    L = count_letters(text) / words * 100
    S = count_sentences(text) / words * 100

    # Calculate index
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # interpret and print result
    print_grade(index)


def count_letters(text):
    '''return number of alphabetic letters in text'''

    letter_count = 0
    for i in range(0, len(text)):
        if text[i].isalpha():
            letter_count += 1

    return letter_count


def count_words(text):
    '''return number of words in text'''

    word_count = 1
    for i in range(0, len(text)):
        if text[i] == " ":
            word_count += 1

    return word_count


def count_sentences(text):
    '''return number of sentences in text'''

    sentence_count = 0
    for i in range(0, len(text)):
        if text[i] in [".", "!", "?"]:
            sentence_count += 1

    return sentence_count


def print_grade(index):
    '''inteprets index and prints grade'''

    if index > 15:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


main()